tests/fsharp/Compiler/Language/InterfaceTests.fs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.UnitTests
+
+open FSharp.Compiler.SourceCodeServices
+open NUnit.Framework
+open FSharp.TestHelpers
+
+[<TestFixture>]
+module InterfaceTests =
+
+    [<Test>]
+    let ShouldWork() =
+        CompilerAssert.Pass
+            """
+type IGet<'T> =
+    abstract member Get : unit -> 'T
+    
+type GetTuple() =
+    interface IGet<int * int> with
+        member x.Get() = 1, 2
+    
+type GetFunction() =
+    interface IGet<unit->int> with
+        member x.Get() = fun () -> 1
+
+type GetAnonymousRecord() =
+    interface IGet<{| X : int |}> with
+        member x.Get() = {| X = 1 |}
+
+type GetNativePtr() =
+    interface IGet<nativeptr<int>> with
+        member x.Get() = failwith "not implemented"
+
+type TUnion = A | B of int
+
+type GetUnion() =
+    interface IGet<TUnion> with
+        member x.Get() = B 2
+
+exit 0
+            """
+
+
+    let multiTypedInterfaceSource = """
+open System
+type AnEnum =
+    | One
+    | Two
+
+type AClass =
+    val value: int
+    
+    new(value) = { value = value }
+
+type IInterface<'a> =
+    abstract GetIt: 'a -> 'a
+
+type implementation () =
+    interface IInterface<bool>       with member _.GetIt(x) = x
```

Comment:
We need to add tests to include (struct)tuples, (struct)records, (struct)unions, (struct)anon records, function as the type argument for the generic interface. We must ensure the names are being generated properly.

Reply:
We also need tests that combine multiple generic interfaces and default interface consumption.

Reply:
We should have tests that assert the name of the methods that get generated; I think the IL verifier will be sufficient for that.

Reply:
@TIHan , okay I will get on that.